specifyweb/stored_queries/execution.py

```diff
@@ -85,6 +85,14 @@ def filter_by_collection(model, query, collection):
         logger.info("filtering geologic time period rank to discipline: %s", collection.discipline.name)
         return query.filter(model.GeologicTimePeriodTreeDefID == collection.discipline.geologictimeperiodtreedef_id)
 
+    if model is models.TectonicUnit:
+        logger.info("filtering tectonic unit to discipline: %s", collection.discipline.name)
+        return query.filter(model.TectonicUnitTreeDefID == collection.discipline.tectonicunittreedef_id)
+
+    if model is models.TectonicUnitTreeDefItem:
```

Comment:
this line prevents from querying for tectonic unit item on form when adding the def item in the relationship